x-pack/plugins/security_solution/public/detections/components/rules/duration_input/index.tsx

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiFieldNumber,
+  EuiFormRow,
+  EuiSelect,
+  EuiFormControlLayout,
+  transparentize,
+} from '@elastic/eui';
+import React, { useCallback } from 'react';
+import styled from 'styled-components';
+
+import type { FieldHook } from '../../../../shared_imports';
+import { getFieldValidityAndErrorMessage } from '../../../../shared_imports';
+import * as I18n from './translations';
+
+interface DurationInputProps {
+  durationValueField: FieldHook<number | undefined>;
+  durationUnitField: FieldHook<string>;
+  minimumValue?: number;
+  isDisabled: boolean;
+  durationUnitOptions?: Array<{ value: 's' | 'm' | 'h' | 'd'; text: string }>;
```

Comment:
Noticed the `Day` unit here and in the translations -- is this just for component re-use and supporting other duration units?

Reply:
Just for reuse options at the moment and making it as similar as possible to ScheduleItem - I wanted them to have the same capabilities so it's easier to start using this one instead of ScheduleItem when we need a duration control.